File: app/src/main/java/com/arctouch/codechallenge/ui/activity/presenter/MovieDetailsActivityPresenter.java
package com.arctouch.codechallenge.ui.activity.presenter;

import android.content.Intent;

import com.arctouch.codechallenge.entity.Movie;

public interface MovieDetailsActivityPresenter {

    void getMovie(int movieId);
    void start(Intent intent);

    interface View {
        void showMovieDetails(Movie movie);
        void updateProgressBar(boolean show);
    }

}

File: app/src/main/java/com/arctouch/codechallenge/ui/activity/presenter/MovieDetailsActivityPresenter.kt
package com.arctouch.codechallenge.ui.activity.presenter

import android.content.Intent

import com.arctouch.codechallenge.entity.Movie

interface MovieDetailsActivityPresenter {

    fun getMovie(movieId: Int)
    fun start(intent: Intent)

    fun clearCompositeDisposable()

    interface View {
        fun showMovieDetails(movie: Movie)
        fun updateProgressBar(show: Boolean)
    }

}
